### PR TITLE
Add delete action in the Component list page

### PR DESCRIPTION
**Pull request by debsmita1**
## Fixes 
https://issues.redhat.com/browse/HAC-544

## Description
- Added Delete action in the kebab menu which on clicked will delete the component CR
- Added tests

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
delete-components

## How to test or reproduce?
Goto: https://prod.foo.redhat.com:1337/beta/hac/app-studio/components?application=<application-name>

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/ComponentListView/ComponentListViewPage.tsx [hunk 1]
```diff
@@ -51,48 +53,95 @@ export const ComponentListViewPage: React.FC<ComponentListViewPageProps> = ({ ap
...
+      : updatedCheckedColumns.add(selectedItem);
+    setSelectedComponents(updatedCheckedColumns);
+  };
+
+  const deleteSelectedRows = async () => {
+    await Promise.all(
+      Array.from(selectedComponents).map((component) =>
+        deleteComponent(component, application, namespace),
+      ),
+    )
```
Comment (rottencandy) [inline src/components/ComponentListView/ComponentListViewPage.tsx:73]: These callbacks are making use of component state, would be good to memoize them.

File: src/utils/delete-utils.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    spec: {
+      application,
+    },
+  };
+  // TODO: Make Api Calls here
+  return k8sDeleteResource({
+    model: ComponentModel,
+    resource: componentData,
+  });
```
Comment (rottencandy) [inline src/utils/delete-utils.ts:33]: Not an issue but seems strange that we have to pass the component's entire spec just to delete it :thinking:

File: src/components/ComponentListView/ComponentListViewPage.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.hacDev-component-list-view-page__trash-icon {
+  color: var(--pf-global--BackgroundColor--dark-400);
+  &:hover {
+    cursor: pointer;
```
Comment (rottencandy) [inline src/components/ComponentListView/ComponentListViewPage.scss:4]: `<Button>` doesn't make it pointer by default?

File: src/utils/delete-utils.ts (hunk 2 above)
Comment (debsmita1) [reply src/utils/delete-utils.ts:30]: removed the spec

File: src/components/ComponentListView/ComponentListViewPage.scss (hunk 3 above)
Comment (debsmita1) [reply src/components/ComponentListView/ComponentListViewPage.scss]: Thanks for catching this. Removed the css

File: src/utils/delete-utils.ts (hunk 2 above)
Comment (rottencandy) [reply src/utils/delete-utils.ts:30]: @debsmita1 we won't need the `application` arg now.

**Review by rottencandy — COMMENTED** [review]
When pressing delete, there is no indication that components are being deleted, maybe we can use `isLoading` for the button to show a spinner while the deletion happens. WDYT?

Also, not sure if it would be good should show a notification once deletion is completed.

https://user-images.githubusercontent.com/20013884/154287585-ae7be36b-d02d-4259-8f82-c6a3d2ce1375.mp4

**Comment by debsmita1** [comment]
> When pressing delete, there is no indication that components are being deleted, maybe we can use `isLoading` for the button to show a spinner while the deletion happens. WDYT?
> 
> Also, not sure if it would be good should show a notification once deletion is completed.
> 
>  0.mp4

^ @beaumorley @rohitkrai03

**Comment by rohitkrai03** [comment]
> > When pressing delete, there is no indication that components are being deleted, maybe we can use `isLoading` for the button to show a spinner while the deletion happens. WDYT?
> > Also, not sure if it would be good should show a notification once deletion is completed.
> > 0.mp4
> 
> ^ @beaumorley @rohitkrai03

@debsmita1 Can you start a conversation around this in slack to get faster turnaround?

**Comment by debsmita1** [comment]
> @debsmita1 You might have to rework this a bit because designs have changes around this page.

@rohitkrai03  Updated. PTAL !

**Review by rottencandy — APPROVED** [review]
/lgtm

**Review by rohitkrai03 — APPROVED** [review]
@debsmita1 Overall looks good but I think we should also add the delete confirmation modal similar to what we have in console. Its not specified in the UX designs but I think its good pattern to follow here. I think we'll need the modal launcher from console but its not in a good state to bring in as is. Maybe we do this as a follow up story.

**Pull request merged**